projects/mCaptcha/project.nix

```
@@ -0,0 +1,10 @@
+{pkgs, ...} @ args: {
+  packages = {inherit (pkgs) mcaptcha mcaptcha-cache;};
+  nixos = {
+    modules.service = ./service.nix;
+    tests = {
+      create-locally = import ./tests/create-locally.nix args;
+      bring-your-own-services = import ./tests/bring-your-own-services.nix args;
```

```suggestion
{pkgs, sources }: {
  packages = {inherit (pkgs) mcaptcha mcaptcha-cache;};
  nixos = {
    modules.service = ./service.nix;
    tests = {
      create-locally = import ./tests/create-locally.nix { sources };
      bring-your-own-services = import ./tests/bring-your-own-services.nix { sources };
```

Explicit > implicit? IDK, maybe not abstracting over `args` will make it scale worse, but it feels from the current POV like we should say what exactly we're passing on. You'd essentially have to look at 3 files otherwise (top-level, this, the test file) to understand what's happening.

It could also be something like  `mapAttrs' (name: dir: ${stripSuffix name} = import dir { sources }; ) (readDir ./tests)`

---

I changed to have explicit arguments, but still using `args`.